quarkus-extension/engine/runtime/src/main/java/org/camunda/bpm/quarkus/engine/extension/impl/ManagedJobExecutor.java

```diff
@@ -68,8 +56,18 @@ protected void startExecutingJobs() {
 
   @Override
   protected void stopExecutingJobs() {
-    // nothing to do, the AcquireJobsRunnable instance will
-    // be stopped when the ManagedExecutor instance is shut down.
+
+    // Ask the thread pool to finish and exit
+    taskExecutor.shutdown();
+
+    // Waits for 1 minute to finish all currently executing jobs
+    try {
+      if(!taskExecutor.awaitTermination(60L, TimeUnit.SECONDS)) {
+        LOG.timeoutDuringShutdown();
+      }
+    } catch (InterruptedException e) {
+      LOG.interruptedWhileShuttingDownjobExecutor(e);
+    }
```

Comment:
I think we still don't need this logic. What changes in the JobExecutor lifecycle when using an independent thread pool?

Reply:
When we use an independent thread pool, we're not using the container-managed executor. Because of this, the container doesn't manage our Executor, i.e., it doesn't shut it down. So we need some logic to make sure that the task executor is shut down correctly.